src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -582,6 +583,62 @@ public ActionResult<Models.DotnetMonitorInfo> GetInfo()
             Utilities.GetProcessKey(pid, uid, name));
         }
 
+        [HttpGet("stacks", Name = nameof(CaptureStacks))]
+        [ProducesWithProblemDetails(ContentTypes.ApplicationJson, ContentTypes.TextPlain)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status429TooManyRequests)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(void), StatusCodes.Status202Accepted)]
+        [RequestLimit(LimitKey = Utilities.ArtifactType_Stacks)]
+        [EgressValidation]
+        public async Task<ActionResult> CaptureStacks(
+            [FromQuery]
+            int? pid = null,
+            [FromQuery]
+            Guid? uid = null,
+            [FromQuery]
+            string name = null,
+            [FromQuery]
+            string egressProvider = null)
+        {
+            ProcessKey? processKey = Utilities.GetProcessKey(pid, uid, name);
+
+            return await InvokeForProcess(async processInfo =>
+            {
+                bool plainText = Request.GetTypedHeaders().Accept?.Contains(TextPlainHeader) ?? false;
+
+                var settings = new EventStacksPipelineSettings
+                {
+                    Duration = Timeout.InfiniteTimeSpan
+                };
+                await using var eventTracePipeline = new EventStacksPipeline(new DiagnosticsClient(processInfo.EndpointInfo.Endpoint),
+                settings);
+
+                Task runPipelineTask = eventTracePipeline.RunAsync(HttpContext.RequestAborted);
```

Comment:
```suggestion
                Task runPipelineTask = await eventTracePipeline.StartAsync(HttpContext.RequestAborted);
```

and remove the `await Task.Yield()`.